packages/front-end/components/Metrics/DateGraph.tsx

```diff
@@ -28,6 +29,95 @@ interface Datapoint {
   oor?: boolean; // out of range
 }
 
+function getDatapointFromDate(date: number, data: Datapoint[]) {
+  // find the closest datapoint to the date
+  const datapoint = data.reduce((acc, cur) => {
+    const curDate = getValidDate(cur.d).getTime();
+    const accDate = getValidDate(acc.d).getTime();
+    return Math.abs(curDate - date) < Math.abs(accDate - date) ? cur : acc;
+  });
+  // if it's within 1 day, return it
+  if (Math.abs(getValidDate(datapoint.d).getTime() - date) < 86400000) {
+    return datapoint;
+  }
+  return null;
+}
+
+function getTooltipDataFromDatapoint(
+  datapoint: Datapoint,
+  data: Datapoint[],
+  innerWidth: number,
+  yScale: ScaleLinear<unknown, unknown, never>
+) {
+  const index = data.indexOf(datapoint);
+  if (index === -1) {
+    return null;
+  }
+  const x = (data.length > 0 ? index / data.length : 0) * innerWidth;
+  const y = (yScale(datapoint.v) ?? 0) as number;
+  return { x, y, d: datapoint };
+}
+
+function getDateFromX(
+  x: number,
+  data: Datapoint[],
+  width: number,
+  marginLeft: number,
+  marginRight: number
+) {
+  const innerWidth = width - marginRight - marginLeft + width / data.length - 1;
+  const px = x / innerWidth;
+  const index = Math.max(
+    Math.min(Math.round(px * data.length), data.length - 1),
+    0
+  );
+  const datapoint = data[index];
+  return getValidDate(datapoint.d).getTime();
+}
+
+function getTooltipContents(
+  d: Datapoint,
+  type: MetricType,
+  method: "sum" | "avg",
+  smoothBy: "day" | "week"
+) {
+  if (!d || d.oor) return null;
+  return (
+    <>
+      {type === "binomial" ? (
+        <div className={styles.val}>
+          <em>n</em>: {Math.round(d.v)}
+          {smoothBy === "week" && (
+            <sub style={{ fontWeight: "normal", fontSize: 8 }}> smooth</sub>
+          )}
+        </div>
+      ) : (
+        <>
+          <div className={styles.val}>
+            {method === "sum" ? `Σ` : `μ`}:{" "}
+            {formatConversionRate(type, d.v as number)}
+            {smoothBy === "week" && (
+              <sub style={{ fontWeight: "normal", fontSize: 8 }}> smooth</sub>
+            )}
+          </div>
+          {"s" in d && method === "avg" && (
+            <div className={styles.secondary}>
+              {`σ`}: {formatConversionRate(type, d.s)}
+              {smoothBy === "week" && (
+                <sub style={{ fontWeight: "normal", fontSize: 8 }}> smooth</sub>
+              )}
+            </div>
+          )}
+          <div className={styles.secondary}>
+            <em>n</em>: {Math.round(d.c)}
+          </div>
+        </>
+      )}
+      <div className={styles.date}>{date(d.d as Date)}</div>
+    </>
+  );
```

Comment:
The tooltip contents are pretty difficult to parse, not sure if it's because of the existing Datapoint and how it has properties like v, s, etc. I think this function probably warrants an explanation comment that maybe includes an example of the output.

Reply:
decided to push this out a bit since it will probably change again once we address ratio metrics.